src/pages/iou/ReceiptSelector/index.js

```diff
@@ -0,0 +1,184 @@
+        }),
+    }),
+
+    /** Holds data related to Money Request view state, rather than the underlying Money Request data. */
+    iou: PropTypes.shape({
+        id: PropTypes.string,
+        amount: PropTypes.number,
+        currency: PropTypes.string,
+        participants: PropTypes.arrayOf(
+            PropTypes.shape({
+                accountID: PropTypes.number,
+                login: PropTypes.string,
+                isPolicyExpenseChat: PropTypes.bool,
+                isOwnPolicyExpenseChat: PropTypes.bool,
+                selected: PropTypes.bool,
+            }),
+        ),
+    }),
+};
+
+const defaultProps = {
+    receiptModal: {
+        isAttachmentInvalid: false,
+        attachmentInvalidReasonTitle: '',
+        attachmentInvalidReason: '',
+    },
+    report: {},
+    route: {
+        params: {
+            iouType: '',
+            reportID: '',
+        },
+    },
+    iou: {
+        id: '',
+        amount: 0,
+        currency: CONST.CURRENCY.USD,
+        participants: [],
+    },
+};
+
+function ReceiptSelector(props) {
+    const reportID = lodashGet(props.route, 'params.reportID', '');
+    const iouType = lodashGet(props.route, 'params.iouType', '');
+    const isAttachmentInvalid = lodashGet(props.receiptModal, 'isAttachmentInvalid', false);
+    const attachmentInvalidReasonTitle = lodashGet(props.receiptModal, 'attachmentInvalidReasonTitle', '');
+    const attachmentInvalidReason = lodashGet(props.receiptModal, 'attachmentInvalidReason', '');
+    const [receiptImageTopPosition, setReceiptImageTopPosition] = useState(0);
+    const {isSmallScreenWidth} = useWindowDimensions();
+    const {translate} = useLocalize();
+    const {isDraggingOver} = useContext(DragAndDropContext);
+
+    /**
+     * Sets the Receipt objects and navigates the user to the next page
+     * @param {Object} file
+     * @param {Object} iou
+     * @param {Object} report
+     */
+    const setReceiptAndNavigate = (file, iou, report) => {
+        if (!ReceiptUtils.validateReceipt(file)) {
+            return;
+        }
+
+        const filePath = URL.createObjectURL(file);
+        IOU.setMoneyRequestReceipt(filePath, file.name);
+        IOU.navigateToNextPage(iou, iouType, reportID, report);
+    };
+
+    return (
+        <View style={[styles.uploadReceiptView(isSmallScreenWidth)]}>
+            {!isDraggingOver ? (
+                <>
+                    <View
+                        onLayout={({nativeEvent}) => {
+                            setReceiptImageTopPosition(PixelRatio.roundToNearestPixel(nativeEvent.layout.top));
+                        }}
+                    >
+                        <ReceiptUpload
+                            width={CONST.RECEIPT.ICON_SIZE}
+                            height={CONST.RECEIPT.ICON_SIZE}
+                        />
+                    </View>
+                    <Text style={[styles.textReceiptUpload]}>{translate('receipt.upload')}</Text>
```

Comment:
Coming from https://github.com/Expensify/App/issues/27551
This was causing the tabs to swipe when selecting the text, since the cursor was outside the text area and it was triggering tab navigation (swipe gesture)
We resolved this by using panHandlers on the parent View (`onPanResponderTerminationRequest: false`, don't allow something else (tab navigation in our case) to take over gestures) and extending text area to take full parent width